### PR TITLE
Add support function to filter authenticators

### PR DESCRIPTION
**Pull request by AfraHussaindeen**
## Purpose
Fix related to  https://github.com/wso2-enterprise/asgardeo-product/issues/20580

Related PRs : 
- https://github.com/wso2/identity-apps/pull/4781

### CODE REVIEW

File: components/org.wso2.carbon.identity.conditional.auth.functions.utils/pom.xml [hunk 1]
```diff
@@ -0,0 +1,80 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+  ~ Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```
Comment (ashendes) [inline components/org.wso2.carbon.identity.conditional.auth.functions.utils/pom.xml:4]: Update all license headers to the new license.

File: components/org.wso2.carbon.identity.conditional.auth.functions.utils/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/utils/FilterAuthenticatorsFunctionImpl.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            String authenticator = option.get(FrameworkConstants.JSAttributes.AUTHENTICATOR);
+
+            if (!excludeAuthenticator.equals(authenticator)) {
+                Map<String, String> idpMap = new HashMap<>();
+                if (FrameworkConstants.LOCAL_IDP_NAME.equals(idp)) {
+                    idpMap.put(FrameworkConstants.JSAttributes.AUTHENTICATOR, authenticator);
+                } else {
+                    idpMap.put(FrameworkConstants.JSAttributes.IDP, idp);
+                }
+                result.put(String.valueOf(index++), idpMap);
```
Comment (sahandilshan) [inline components/org.wso2.carbon.identity.conditional.auth.functions.utils/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/utils/FilterAuthenticatorsFunctionImpl.java:48]: Why can't we pass the integer value as it is here? Why do we need to make this value string?

Comment (AfraHussaindeen) [reply components/org.wso2.carbon.identity.conditional.auth.functions.utils/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/utils/FilterAuthenticatorsFunctionImpl.java:48]: As in https://github.com/wso2/carbon-identity-framework/blob/1ceee196051722474017ab432cee84d4a949b7a3/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/JsNashornGraphBuilder.java#L371C27-L371C61, the expected return format is `Map<String, Map<String, String>>`

File: components/org.wso2.carbon.identity.conditional.auth.functions.utils/pom.xml (hunk 1 above)
Comment (AfraHussaindeen) [reply components/org.wso2.carbon.identity.conditional.auth.functions.utils/pom.xml]: Resolved via https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/148/commits/9b2f556bc689b5f024cb8bb6daf372b86d7abf46

File: components/org.wso2.carbon.identity.conditional.auth.functions.utils/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/utils/FilterAuthenticatorsFunctionImpl.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                                                                 String excludeAuthenticator) {
+
+        Map<String, Map<String, String>> result = new HashMap<>();
+        int index = 0;
+
+        for (Map<String, String> option : authenticatorOptions) {
+            String idp = option.get(FrameworkConstants.JSAttributes.IDP);
+            String authenticator = option.get(FrameworkConstants.JSAttributes.AUTHENTICATOR);
+
+            if (!excludeAuthenticator.equals(authenticator)) {
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.conditional.auth.functions.utils/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/utils/FilterAuthenticatorsFunctionImpl.java:41]: Shall we use `Stringutils.equals` to make sure this is null safe.

Comment (AfraHussaindeen) [reply components/org.wso2.carbon.identity.conditional.auth.functions.utils/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/utils/FilterAuthenticatorsFunctionImpl.java]: Resolved via https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/148/commits/08e6d85667e7f226e9f1914e63394c90c7512618

**Review by ashensw — APPROVED** [review]

**Pull request merged**